### ISSUE TITLE
Pad nested callbacks with newlines

### ISSUE THREAD

**Issue by brenr** [eslint/eslint#13535]
**Please describe what the rule should do:**
Pad large callbacks with new lines.

**What new ECMAScript feature does this rule relate to?**
Not applicable - general formatting.

**What category of rule is this? (place an "X" next to just one item)**
[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
const assert = require('assert');
const pg = require('pg');
const DBManager = require('../../src/database/DBManager.js');

describe('DBManager', () => {
// pad with new line here
    describe('#get', () => {
// pad with new line here
        it('should return an instance of pg.Pool', done => {
            assert.equal(true, DBManager.get() instanceof pg.Pool);
            done();
        });
// pad with new line here
    });
// pad with new line here
});
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It helps break up large function callback nesting as seen in mocha tests.

**Are you willing to submit a pull request to implement this rule?**
I'm juggling a handful of projects currently, so unfortunately no.

**Comment by brenr** [comment]
> Thanks for the issue.
> 
> Can you check with [padded-blocks](https://eslint.org/docs/rules/padded-blocks).
> 
> [demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyplc2xpbnQgcGFkZGVkLWJsb2NrczogW1wiZXJyb3JcIiwgXCJhbHdheXNcIl0qL1xuY29uc3QgYXNzZXJ0ID0gcmVxdWlyZSgnYXNzZXJ0Jyk7XG5jb25zdCBwZyA9IHJlcXVpcmUoJ3BnJyk7XG5jb25zdCBEQk1hbmFnZXIgPSByZXF1aXJlKCcuLi8uLi9zcmMvZGF0YWJhc2UvREJNYW5hZ2VyLmpzJyk7XG5cbmRlc2NyaWJlKCdEQk1hbmFnZXInLCAoKSA9PiB7XG5cbiAgICBkZXNjcmliZSgnI2dldCcsICgpID0+IHtcblxuICAgICAgICBpdCgnc2hvdWxkIHJldHVybiBhbiBpbnN0YW5jZSBvZiBwZy5Qb29sJywgZG9uZSA9PiB7XG4gICAgICAgICAgXG4gICAgICAgICAgYXNzZXJ0LmVxdWFsKHRydWUsIERCTWFuYWdlci5nZXQoKSBpbnN0YW5jZW9mIHBnLlBvb2wpO1xuICAgICAgICAgIFxuICAgICAgICAgIGRvbmUoKTtcbiAgICAgICAgICBcbiAgICAgICAgfSk7XG5cbiAgICB9KTtcblxufSk7Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo5LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19)

That applies to all blocks, so it would pad `if`, `else`, `try`, `catch`, etc.

**Comment by anikethsaha** [comment]
Can you explain with a bit more details? 

are you looking for only with callbacks (function in general then)?

**Comment by brenr** [comment]
> Can you explain with a bit more details?
> 
> are you looking for only with callbacks (function in general then)?

Yes, correct. Maybe it can be identified by either single and multi line functions?

**Comment by anikethsaha** [comment]
IMO, 

It looks like an enhancement for [padding-line-between-statements](https://eslint.org/docs/rules/padding-line-between-statements) as it doesn't support `ArrowFunctionExpression` and `FunctionExpression` (I think, not sure tho)

So as per the new [policy](https://eslint.org/blog/2020/05/changes-to-rules-policies) stylistic rules are frozen for enhancement and this is an enhancement added to a stylistic rule

Maybe a good option for a custom rule/plugin

**Comment by mdjermanovic** [comment]
@brenr thanks for the rule proposal!

Given the [changes to our rules policy](https://eslint.org/blog/2020/05/changes-to-rules-policies), I'd suggest making a custom rule, which could be based on the `padded-blocks` rule.
